content/docs/rfcs/34/README.md

```diff
@@ -0,0 +1,123 @@
+---
+slug: 34
+title: 34/WAKU2-PEER-EXCHANGE
+name: Waku v2 Peer Exchange
+status: raw
+category: Standards Track
+tags: waku/core-protocol
+editor: Daniel Kaiser <danielkaiser@status.im>
+contributors:
+---
+
+# Abstract
+
+This document specifies a simple request-response peer exchange protocol.
+Responders send information about a requested number of peers.
+The main purpose of this protocol is providing resource restricted devices with peers.
+
+# Background and Motivation
+
+Resource restricted devices may not feasibly be able to take part in distributed random sampling ambient peer discovery protocols such as [33/WAKU2-DISCV5](https://rfc.vac.dev/spec/33/).
+The Waku peer discovery protocol specified in this document allows resource restricted devices to request a list of peers from a (stronger) peer.
+Network parameters necessary to connect to this stronger peer can, for instance, be learned from a static bootstrapping method.
+The advantage of using Waku peer exchange to discover new peers over simply adding static bootstrap peers to the peer list is a more even load distribution.
+If a lot of (resource restricted) nodes would use static bootstrap nodes as relay or store nodes, the load on these would be very high.
+Heavily used static nodes also add a centralized element. Downtime of such a node might significantly impact the network.
+
+However, the resource efficiency of this protocol comes at an anonymity cost, which is explained in the [Security/Privacy Considerations](#securityprivacy-considerations) section.
+This protocol SHOULD only be used if [33/WAKU2-DISCV5](https://rfc.vac.dev/spec/33/) is infeasible.
+
+# Theory and Protocol Semantics
+
+The peer exchange protocol specified in this document is a simple request-response protocol.
+As Figure 1 illustrates, the requesting node sends a request to a peer, which acts as the responder.
+The responder replies with a list of [multiaddresses](https://docs.libp2p.io/concepts/addressing/).
```

Comment:
I will look into this.
It should be in line with the [StrictNoSign Policy](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#why-are-we-using-the-strictnosign-signature-policy).
Also have to check discv5 ENRs regarding this.

Reply:
[Waku ENR](https://rfc.vac.dev/spec/31/) payload is derived from [EIP-778](https://eips.ethereum.org/EIPS/eip-778).

It contains a `signature`of1 the content, signed by the public key encoded in the `secp256k1` field (signature is defined by identity scheme specified in `id` field).

Do we need a similar mechanism for this protocol? Should we simply return the ENRs to include this security feature?

[EIP-778](https://eips.ethereum.org/EIPS/eip-778) does not specify why the signature is provided. After all, if the details are incorrect then connection would just failed because remote peer id must back the public key in the ENR.

Reply:
> StrictNoSign Policy.

This only applies to gossipsub to avoid linking a message to it's originator
Here, a peer is expected to announce him himself on the network, so we already know he is the originator of this data

Reply:
Yes. I was looking for implications I might have missed :sweat_smile: . Did not find any problems so far.
I'm switching to ENRs in the response.

I'll add your statement to the security section to make that explicit.

Reply:
Still, ENRs could be used to track peers over multiple "sessions", and maybe used to facilitate some linking attacks.
Ill mention that in security section. I'll look more into that when I continue with the anonymity analysis.

Reply:
Yes, thought this is more a discv5 issue than a PX issue

See https://github.com/status-im/nim-libp2p/issues/727 for PeerId (which is the identity used in the ENR) rotation discussion
(TLDR: we should change peerid when we change IP, which is not trivial to do)

Reply:
Yes. I remeber the discord convo :).
I touched on possible implications in the security section.